llvm/tools/llvm-gpu-loader/CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+set(LLVM_LINK_COMPONENTS
+  BinaryFormat
+  Object
+  Option
+  Support
+  FrontendOffloading
+)
+
+add_llvm_tool(llvm-gpu-loader
+  llvm-gpu-loader.cpp
+
+  # TODO: We intentionally split this currently due to statically linking the
+  #       GPU runtimes. Dynamically load the dependencies, possibly using the
+  #       LLVM offloading API when it is complete.
+  PARTIAL_SOURCES_INTENDED
```

Comment:
This is not a correct use of `PARTIAL_SOURCES_INTENDED`. It is intended to be used only when you are building multiple targets within a single file. Here, you should just include all sources and use `#ifdef`.